includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -863,6 +870,17 @@ public function process_payment( $order_id ) {
 		$order = wc_get_order( $order_id );
 
 		try {
+			$fraud_prevention_service = Fraud_Prevention_Service::instance();
+			if ( $fraud_prevention_service->is_enabled() ) {
+				// phpcs:ignore WordPress.Security.NonceVerification.Missing
+				if ( ! $fraud_prevention_service->verify_token( $_POST ) ) {
```

Comment:
Let's also dispatch here the token value instead of complete array.

Reply:
From one side I also didn't like the idea of passing whole `$_POST` but if we pass token value then we need to pull out the `isset( $post_data['wcpay-fraud-prevention-token'] )` into something like
```
if ( isset( $_POST[Fraud_Prevention_Service::REQUEST_PARAM] && $fraud_prevention_service->verify_token( $_POST[ Fraud_Prevention_Service::REQUEST_PARAM ] ) )
```
which to me looks like something that should be refactored into one method.

Reply:
Not really: `->verify_token ( $POST[...] ?? null)`. And verify token should ensure null is not accepted as a valid value.